Octokit.Tests.Integration/Clients/OrganizationSecretsClientTests.cs

```diff
@@ -0,0 +1,259 @@
+
+                Assert.NotEmpty(visibilityRepos.Repositories);
+                Assert.Equal(2, visibilityRepos.Count);
+            }
+#endif
+        }
+
+        public class RemoveRepoFromOrganizationSecretMethod
+        {
+#if SODIUM_CORE_AVAILABLE
+            [OrganizationTest]
+            public async Task RemoveSelectedRepositoriesForSecret()
+            {
+                var github = Helper.GetAuthenticatedClient();
+
+                var secretName = "REMOVE_SELECTED_REPO_TEST";
+
+                var repo1 = await CreateRepoIfNotExists(github, "remove-secrets-selected-repo-test-1");
+                var repo2 = await CreateRepoIfNotExists(github, "remove-secrets-selected-repo-test-2");
+
+                var key = await github.Organization.Actions.Secrets.GetPublicKey(ORG);
+                var upsertSecret = GetSecretForCreate("secret", key, new Repository[] { repo1, repo2 });
+                await github.Organization.Actions.Secrets.CreateOrUpdate(ORG, secretName, upsertSecret);
+
+                await github.Organization.Actions.Secrets.RemoveRepoFromOrganizationSecret(ORG, secretName, repo2.Id);
+
+                var visibilityRepos = await github.Organization.Actions.Secrets.GetSelectedRepositoriesForSecret(ORG, secretName);
+
+                Assert.NotEmpty(visibilityRepos.Repositories);
+                Assert.Equal(1, visibilityRepos.Count);
+            }
+#endif
+        }
+
+#if SODIUM_CORE_AVAILABLE
+        private static UpsertOrganizationSecret GetSecretForCreate(string secretValue, SecretsPublicKey key)
+        {
+            var secretBytes = Encoding.UTF8.GetBytes(secretValue);
+            var publicKey = Convert.FromBase64String(key.Key);
+            var sealedPublicKeyBox = SealedPublicKeyBox.Create(secretBytes, publicKey);
+
+            var upsertValue = new UpsertOrganizationSecret
+            {
+                EncryptedValue = Convert.ToBase64String(sealedPublicKeyBox),
+                KeyId = key.KeyId,
+                Visibility = "all"
+
+            };
+
+            return upsertValue;
+        }
+
+        private static UpsertOrganizationSecret GetSecretForCreate(string secretValue, SecretsPublicKey key, Repository[] repos)
+        {
+            var secretBytes = Encoding.UTF8.GetBytes(secretValue);
+            var publicKey = Convert.FromBase64String(key.Key);
+            var sealedPublicKeyBox = SealedPublicKeyBox.Create(secretBytes, publicKey);
+
+            var upsertValue = new UpsertOrganizationSecret
+            {
+                EncryptedValue = Convert.ToBase64String(sealedPublicKeyBox),
+                KeyId = key.KeyId,
+                Visibility = "selected",
+                SelectedRepositoriesIds = repos.Select(r => r.Id)
+
+            };
+
+            return upsertValue;
+        }
+#endif
+
+        private static async Task<Repository> CreateRepoIfNotExists(IGitHubClient github, string name)
```

Comment:
As an alternative for this, there are existing test helpers to create a new repository with a timestamp suffix to prevent clashing:

```
var repoName = Helper.MakeNameWithTimestamp("public-repo");
var context = await github.CreateRepositoryContext(new NewRepository(repoName));
```

This lends itself to creating and tearing down repositories within the context of the test, rather than potentially clobbering existing data.

Reply:
this should be in the new commit